src/assets/LESSAsset.js

```diff
@@ -80,17 +83,65 @@ function getFileManager(less, options) {
       return false;
     }
 
-    async loadFile(filename, currentDirectory) {
-      filename = parseCSSImport(filename);
-      let resolved = await resolver.resolve(
+    async loadFile(filename, currentDirectory, options) {
+      let resolved = await this.findResolution(
         filename,
-        path.join(currentDirectory, 'index')
+        currentDirectory,
+        options
       );
+
       return {
         contents: await fs.readFile(resolved.path, 'utf8'),
         filename: resolved.path
       };
     }
+
+    async findResolution(filename, currentDirectory, options) {
+      const filenamesTried = [];
+      let paths = path.isAbsolute(filename) ? [] : [currentDirectory];
+
+      if (options.paths) {
+        paths = paths.concat(options.paths);
+      }
+
+      if (!path.isAbsolute(filename) && !paths.includes('.') === -1) {
+        paths.push('.');
+      }
+
+      // promise is guaranteed to be asyncronous
+      // which helps as it allows the file handle
+      // to be closed before it continues with the next file
+      return new Promise(function(resolve, reject) {
+        (async function tryPathIndex(i) {
+          if (i < paths.length) {
+            let fullFilename = resolver.resolveFilename(
+              parseCSSImport(filename),
+              paths[i]
+            );
+            try {
+              const resolution = await resolver.resolve(fullFilename, paths[i]);
+              if (resolution) {
+                resolve(resolution);
+              } else {
+                throw new Error();
+              }
+            } catch (err) {
+              filenamesTried.push(fullFilename);
+              tryPathIndex(i + 1);
+            }
+          } else {
+            reject({
```

Comment:
You can just use await and use `throw new Error(...)` or `throw {// the object}` in the catch section